### ISSUE TITLE
Support Edit & Continue

### ISSUE THREAD

**Issue by WaynePhillipsEA** [WaynePhillipsEA/twinbasic#28]
**Is your feature request related to a problem? Please describe.**
twinBASIC should allow edit-and-continue hot patching, as many VB6/VBA developers use this feature.

**Describe the solution you'd like**
When editing source code, the compiler should patch existing running compiled code to match the edited code, without requiring to abort the threads of execution.

**Additional context**
There are limitations, such as edits must not change the stack frame size, so for example you can't make an edit that introduces a new temporary variable. VB6/VBA also has the same restrictions, and requires aborting the execution in that scenario.

**Comment by EduardoVB** [comment]
IMO this feature has huge importance.

**Comment by jpbro** [comment]
Edit & Continue is a huge advantage of VB programming IMO, so I agree with Eduardo that this is an important feature. Would be nice it tB didn't have at least some of the limitations of VB6 in this regard though:

1) Changing/Adding/Removing cases in a Select Case block
2) Adding/Removing Line Continuations

Both of those require aborting execution in VB6. Maybe there's a good technical reason for that, but maybe those cases were just oversights?

One other thing that would be nice - when you introduce a change that requires abort, VB6 pops up a window confirming that you want to reset:

![image](https://user-images.githubusercontent.com/5287585/114937686-18ee5200-9e0c-11eb-9b24-53244cb02635.png)

Pressing Enter will "click" the OK button and reset the project. If you are subclassing or working on highly experimental stuff, this can sometimes crash the IDE, causing you to lose work. If you are typing quickly and unaware that the dialog is about to pop up, it's not uncommon to press Enter and blow up. So while mapping Enter to the default button is normally good, in this case it is potentially very bad. It's understandable if you want to keep the Enter key mapped to the OK button for accessibility, but perhaps there could be a delay (1 or 2 seconds would probably be enough) before Enter will click OK?

**Comment by EduardoVB** [comment]
Yes, also if you want to change local variable names you need to restart. If that could be possible to do without restarting it would be great too.
